Userland/Libraries/LibGfx/LunaWindowTheme.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2023, MacDue <macdue@dueutil.tech>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Array.h>
+#include <LibGfx/CharacterBitmap.h>
+#include <LibGfx/Gradients.h>
+#include <LibGfx/LunaWindowTheme.h>
+#include <LibGfx/Painter.h>
+#include <LibGfx/Palette.h>
+#include <LibGfx/StylePainter.h>
+
+namespace Gfx {
+
+// TODO: Somehow allow colors to be configured in the theme .ini file.
+static Array const s_luna_title_gradient {
+    ColorStop { Color(9, 151, 255), 0.00f },
+    ColorStop { Color(0, 83, 238), 0.14f },
+    ColorStop { Color(0, 80, 238), 0.40f },
+    ColorStop { Color(0, 102, 255), 0.88f },
+    ColorStop { Color(0, 102, 255), 0.93f },
+    ColorStop { Color(0, 91, 255), 0.95f },
+    ColorStop { Color(0, 61, 215), 0.96f },
+    ColorStop { Color(0, 61, 215), 1.00f }
+};
+
+static Array const s_button_gradient_base {
+    ColorStop { Color(72, 146, 247), 0.0f },
+    ColorStop { Color(57, 128, 244), 0.05f },
+    ColorStop { Color(57, 128, 244), 1.0f }
+};
+
+static Array const s_button_gradient_overlay {
+    ColorStop { Color(109, 164, 246), 0.0f },
+    ColorStop { Color::Transparent, 0.05f },
+    ColorStop { Color::Transparent, 1.0f }
+};
+
+static constexpr struct {
+    Color base { 22, 39, 213 };
+    Color middle_shade { 22, 80, 217 };
+    Color light_shade { 32, 102, 234 };
+} s_frame_colors;
+
+static constexpr struct {
+    Color border { 38, 83, 174 };
+} s_button_colors;
+
+static constexpr Gfx::CharacterBitmap s_window_border_radius_mask {
+    "#####"
+    "###  "
+    "##   "
+    "#    "
+    "#    "sv,
+    5, 5
+};
+
+static constexpr Gfx::CharacterBitmap s_window_border_radius_accent {
+    "     "
+    "   ##"
+    "  #  "
+    " #   "
+    " #   "sv,
+    5, 5
+};
+
+IntRect LunaWindowTheme::titlebar_rect(WindowType window_type, WindowMode window_mode, IntRect const& window_rect, Palette const& palette) const
```

Comment:
Should probably rename the class to match the new name too. (just "PlasticWindowTheme" is fine if you think it's too long otherwise)

(also in icon file names, theme name, etc)

Reply:
Done :+1: